crates/solver/src/liquidity/slippage.rs

```diff
@@ -170,11 +170,46 @@ pub struct SlippageContext<'a> {
 }
 
 impl<'a> SlippageContext<'a> {
-    /// Computes the AMM execution maximum input amount.
-    pub fn execution_input_max(&self, input: (H160, U256)) -> Result<(H160, U256)> {
-        let (token, amount) = input;
-        let slippage = self.calculator.compute(self.prices, token, amount)?;
-        Ok((token, slippage.add_to_amount(amount)))
+    /// Applies slippage to the specified AMM execution.
+    pub fn apply_to_amm_execution(
+        &self,
+        mut execution: AmmOrderExecution,
+    ) -> Result<AmmOrderExecution> {
+        let slippage = if let Ok(slippage) =
+            self.calculator
+                .compute(self.prices, execution.input_max.0, execution.input_max.1)
```

Comment:
nit: Not that it should be part of this PR, but I think we should refactor `AmmOrderExecution` to contain `TokenAmount` instead of `(H160, U256)`, since the latter one kills the readability in functions like this one... I need to check what `input_max.0` and `input_max.1` are to understand the context.

Reply:
This is a great suggestion! Will make an issue to keep it in the backlog of "good first issues" that can be easily picked up.

Reply:
https://github.com/cowprotocol/services/issues/692